cinema/serializers.py

```diff
@@ -68,13 +81,75 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
+        )
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "movie_session", "row", "seat")
+        validators = [
+            UniqueTogetherValidator(
+                queryset=Ticket.objects.all(),
+                fields=["movie_session", "row", "seat"]
+            )
+        ]
+
+    def validate(self, attrs):
+        Ticket.validate_seat_and_row(
+            attrs["seat"],
+            attrs["row"],
+            attrs["movie_session"].cinema_hall,
```

Comment:
In the `validate` method of `TicketSerializer`, the `attrs['movie_session']` might not be available directly if `movie_session` is a read-only field. You should ensure that `movie_session` is included in the `attrs` or handle it appropriately to avoid potential `KeyError`.